packages/sdk/src/reliable_channel/reliable_channel.ts

```diff
@@ -595,96 +613,176 @@ export class ReliableChannel<
+          messageId,
+          retrievalHint
+        );
       }
+    }
+  }
+
+  private onMessageLost(event: CustomEvent<HistoryEntry[]>): void {
+    this.syncStatus.onMessagesLost(...event.detail.map((m) => m.messageId));
+  }
+
+  private onQueryOnConnect(event: CustomEvent<IDecodedMessage[]>): void {
+    void this.processIncomingMessages(event.detail);
+  }
+
+  private addEventListeners(): void {
+    this.messageChannel.addEventListener(
+      MessageChannelEvent.OutMessageSent,
+      this.onMessageSent.bind(this)
     );
 
     this.messageChannel.addEventListener(
       MessageChannelEvent.OutMessageAcknowledged,
-      (event) => {
-        if (event.detail) {
-          this.safeSendEvent("message-acknowledged", {
-            detail: event.detail
-          });
-
-          // Stopping retries
-          this.retryManager?.stopRetries(event.detail);
-        }
-      }
+      this.onMessageAcknowledged.bind(this)
     );
 
     this.messageChannel.addEventListener(
       MessageChannelEvent.OutMessagePossiblyAcknowledged,
-      (event) => {
-        if (event.detail) {
-          this.safeSendEvent("message-possibly-acknowledged", {
-            detail: {
-              messageId: event.detail.messageId,
-              possibleAckCount: event.detail.count
-            }
-          });
-        }
-      }
+      this.onMessagePossiblyAcknowledged.bind(this)
     );
 
     this.messageChannel.addEventListener(
       MessageChannelEvent.InSyncReceived,
-      (_event) => {
-        // restart the timeout when a sync message has been received
-        this.restartSync();
-      }
+      this.onSyncMessage.bind(this)
     );
 
     this.messageChannel.addEventListener(
       MessageChannelEvent.InMessageReceived,
-      (event) => {
-        // restart the timeout when a content message has been received
-        if (isContentMessage(event.detail)) {
-          // send a sync message faster to ack someone's else
-          this.restartSync(0.5);
-        }
-      }
+      this.onMessageReceived.bind(this)
     );
 
     this.messageChannel.addEventListener(
       MessageChannelEvent.OutMessageSent,
-      (event) => {
-        // restart the timeout when a content message has been sent
-        if (isContentMessage(event.detail)) {
-          this.restartSync();
-        }
-      }
+      this.onMessageSent.bind(this)
     );
 
     this.messageChannel.addEventListener(
       MessageChannelEvent.InMessageMissing,
-      (event) => {
-        for (const { messageId, retrievalHint } of event.detail) {
-          if (retrievalHint && this.missingMessageRetriever) {
-            this.missingMessageRetriever.addMissingMessage(
-              messageId,
-              retrievalHint
-            );
-          }
-        }
-      }
+      this.onMessageMissing.bind(this)
+    );
+
+    this.messageChannel.addEventListener(
+      MessageChannelEvent.InMessageLost,
+      this.onMessageLost.bind(this)
```

Comment:
nit: usually in constructor you write `this.onMessageLost = this.onMessageLost.bind(this);` and then everywhere you have correct binding

Reply:
or there is a syntax in TypeScript for it

Reply:
Resolved as I removed this change and adopted the syntax for the clean up PR https://github.com/waku-org/js-waku/pull/2733